lib/ansible/executor/powershell/async_wrapper.ps1

```diff
@@ -145,11 +145,14 @@ try {
     $result_json = ConvertTo-Json -InputObject $result -Depth 99 -Compress
     Set-Content $results_path -Value $result_json
 
-    Write-AnsibleLog "INFO - waiting for async process to connect to named pipe for 5 seconds" "async_wrapper"
+    $np_timeout = $Payload.async_named_pipe_timeout * 1000
+    Write-AnsibleLog "INFO - waiting for async process to connect to named pipe for $np_timeout milliseconds" "async_wrapper"
     $wait_async = $pipe.BeginWaitForConnection($null, $null)
-    $wait_async.AsyncWaitHandle.WaitOne(5000) > $null
+    $wait_async.AsyncWaitHandle.WaitOne($np_timeout) > $null
     if (-not $wait_async.IsCompleted) {
-        throw "timeout while waiting for child process to connect to named pipe"
+        throw """Ansible encountered a timeout while waiting for the async task to start and connect to the named
```

Comment:
I think doing it this way preserves the whitespace at the beginning of the line making the actual error message look a bit weird. The only "sane" way I know to get around the line length issue is to just append a string like so

```
$msg = "Ansible encountered a timeout while waiting for the async task to start and connect to the named pipe. "
$msg += "This can be affected by the performance of the target and the timeout can be increased with the "
$msg += "DEFAULT_WIN_ASYNC_STARTUP_TIMEOUT config option"
throw $msg
```

Also the config option stated wasn't correct, you should use the actual key in `base.yml` as that is how they are documented at https://docs.ansible.com/ansible/latest/reference_appendices/config.html#ansible-configuration-settings.

Reply:
Yeah, I got a bit lost with what the config parameter was actually called.